lib/pxenforcer.js

```diff
@@ -153,6 +159,20 @@ class PxEnforcer {
         }
     }
 
+    _normalizeUrl(originalUrl) {
+        const url = new URL(originalUrl);
+        let decodedUrl;
+        if (this._config.URL_DECODE_RESERVED_CHARACTERS) {
+            try {
+                decodedUrl = new URL(`${url.origin}${decodeURIComponent(url.pathname)}${url.search}`);
+            } catch (e) {
+                this.logger.debug(`unable to URL decode reserved characters: ${e}`);
+            }
+        }
+        url.pathname = url.pathname.replace(/\/+$/, '').replace(/\/+/g, '/');
+        return decodedUrl || url;
```

Comment:
This is confusing to me -- are we working with `url` or `decodedUrl`? Why do we have 2 variables that represent this?

What about something more like this?

```js
    _normalizeUrl(originalUrl) {
        let decodedUrl = new URL(originalUrl);
        if (this._config.URL_DECODE_RESERVED_CHARACTERS) {
            try {
                decodedUrl = new URL(`${url.origin}${decodeURIComponent(url.pathname)}${url.search}`);
            } catch (e) {
                this.logger.debug(`unable to URL decode reserved characters: ${e}`);
            }
        }
        decodedUrl.pathname = url.pathname.replace(/\/+$/, '').replace(/\/+/g, '/');
        return decodedUrl;
```

Reply:
@EldarHuman Thoughts on this?